src/app/utils.ts

```diff
@@ -26,23 +28,28 @@ export const actionHook = <C extends AnyActionCreator>(actionCreator: C, body: A
       const result = next(action);
 
       if (matches(action)) {
-        const catchError = (e: Error) => {
-          Sentry.captureException(e);
-          stateHelpers.dispatch(recordError(e));
-        };
         try {
           const promise = boundHook(action);
           if (promise) {
-            services.promiseCollector.add(promise.catch(catchError));
+            services.promiseCollector.add(promise.catch((e) => catchError(e, stateHelpers.dispatch)));
```

Comment:
you could curry this ahead of time so you don't have to pass it in two places

```
const catchError = makeCatchError(stateHelpers);
```

```suggestion
            services.promiseCollector.add(promise.catch(catchError));
```